### PR TITLE
 chore: use google eslint rules for consistency

### PR DESCRIPTION
**Pull request by MarshallOfSound**
Adopt google's JS rules https://github.com/google/eslint-config-google

Notes: no-notes

cc @nornagon

### CODE REVIEW

**Comment by MarshallOfSound** [comment]
The second commit is just the addition of semi-colons, reccomend reviewing by viewing just the first commit with whitespace changes turned off

https://github.com/electron/electron/pull/15582/commits/a585ae7414085f857c806129e63c460ca83196cb?utf8=%E2%9C%93&diff=split&w=1

File: lib/browser/api/auto-updater/squirrel-update-win.js [hunk 1]
```diff
@@ -17,8 +17,8 @@ const isSameArgs = (args) => args.length === spawnedArgs.length && args.every((e
 
 // Spawn a command and invoke the callback when it completes with an error
 // and the output from standard out.
-const spawnUpdate = function (args, detached, callback) {
-  let error, errorEmitted, stderr, stdout
+const spawnUpdate = function(args, detached, callback) {
+  let error; let errorEmitted; let stderr; let stdout
```
Comment (nornagon) [inline lib/browser/api/auto-updater/squirrel-update-win.js]: This change is surprising; multiple `let` seems ok to me? is it a google style guide thing? is there something bad about multiple `let`?

File: lib/browser/api/browser-window.js [hunk 2]
```diff
@@ -82,9 +82,15 @@ BrowserWindow.prototype._init = function () {
   // Though this hack is only needed on macOS when the app is launched from
   // Finder, we still do it on all platforms in case of other bugs we don't
   // know.
-  this.webContents.once('load-url', function () {
-    this.focus()
-  })
+  this.webContents.once(
+      'load-url',
+      /**
+     * @this WebContents
```
Comment (nornagon) [inline lib/browser/api/browser-window.js]: ?

File: lib/browser/api/menu-utils.js [hunk 3]
```diff
@@ -120,10 +120,10 @@ function sortItemsInGroup (group) {
   })
 
   const sortedNodes = sortTopologically(originalOrder, edges)
-  return sortedNodes.map(i => group[i])
+  return sortedNodes.map((i) => group[i])
```
Comment (nornagon) [inline lib/browser/api/menu-utils.js]: i'm a bit sad about losing no-paren functions

File: lib/browser/api/menu.js [hunk 4]
```diff
@@ -222,7 +222,12 @@ function removeExtraSeparators (items) {
   return ret
 }
 
-function insertItemByType (item, pos) {
+/**
+ * @this Menu
```
Comment (nornagon) [inline lib/browser/api/menu.js]: ?

File: lib/browser/api/web-contents.js [hunk 5]
```diff
@@ -145,12 +145,20 @@ const webFrameMethods = [
...
 ]
 
-const asyncWebFrameMethods = function (requestId, method, callback, ...args) {
+/**
+ * @this WebContents
+ * @param {number} requestId
+ * @param {string} method
+ * @param {Function} callback
+ * @param  {...any} args
+ * @return {Promise<any>}
```
Comment (nornagon) [inline lib/browser/api/web-contents.js]: i'll stop putting comments on these odd doc-comments now

**Comment by deepak1556** [comment]
Can the first commit be broken down further, a commit for each config change in `.eslintrc.json` ? It would really make review easier. Thanks!

**Comment by nornagon** [comment]
... any chance of a backport to 4-0-x? this is gonna generate mad conflicts

File: lib/browser/api/auto-updater/squirrel-update-win.js (hunk 1 above)
Comment (MarshallOfSound) [reply lib/browser/api/auto-updater/squirrel-update-win.js]: @nornagon This is a result of running `lint:js --fix`, this is a result of the [`one-var`](https://eslint.org/docs/rules/one-var) rule.  Set here in [google's eslint config](https://github.com/google/eslint-config-google/blob/master/index.js#L277-L281)

File: lib/browser/api/browser-window.js (hunk 2 above)
Comment (MarshallOfSound) [reply lib/browser/api/browser-window.js]: Without declaring what `this` is, eslint rules don't think this non-class function has a `this` scope that is usable.  All the `@this` declarations in JSDoc are for this purpose 👍

File: lib/browser/api/menu-utils.js (hunk 3 above)
Comment (MarshallOfSound) [reply lib/browser/api/menu-utils.js]: It makes diffs cleaner in the long run though 😄

File: lib/browser/api/web-contents.js (hunk 5 above)
Comment (MarshallOfSound) [reply lib/browser/api/web-contents.js]: This function requires the `@this` JS Doc comment, and there is another rule requiring all JS Doc comments to be valid and complete.  So if I add `@this` I also needed to add all the other `@blah` things

**Comment by MarshallOfSound** [comment]
> Can the first commit be broken down further, a commit for each config change in .eslintrc.json ? It would really make review easier. Thanks!

Not easily, the changes in the eslintrc.json file aren't whats causing the most changes.  99% of these changes come from just enabling Google's eslint config.  Doing that part by part is quite tricky / not possible.

**Comment by MarshallOfSound** [comment]
> ... any chance of a backport to 4-0-x? this is gonna generate mad conflicts

Yeah I can do that once this is approved / merged 👍 Will defs be manual but will result in less conflicts 😄

**Comment by deepak1556** [comment]
So now that the style matches google, I believe we could use clang-format for js ? linter and formatter might have some conflicts in that case, not sure.

File: lib/browser/api/auto-updater/squirrel-update-win.js (hunk 1 above)
Comment (ckerr) [reply lib/browser/api/auto-updater/squirrel-update-win.js]: The autofixed code is worse though.

**Comment by ckerr** [comment]
>> ... any chance of a backport to 4-0-x? this is gonna generate mad conflicts
>
> Yeah I can do that once this is approved / merged ￼ Will defs be manual but will result in less conflicts ￼

A 20,000 line diff into a stabilization branch seems like an insanely bad idea.

File: lib/browser/api/auto-updater/squirrel-update-win.js (hunk 1 above)
Comment (nornagon) [reply lib/browser/api/auto-updater/squirrel-update-win.js]: I think this autofixed code could be improved by moving the `let` declarations to the point at which the variables are actually used. I'm OK with this lint rule staying as is.

**Comment by nornagon** [comment]
> A 20,000 line diff into a stabilization branch seems like an insanely bad idea.

it's not comparable with a 20,000 line diff of hand-written code. this is more like a 5-line diff in config that just happens to be spread across 20,000 lines of actual code.

File: lib/browser/api/browser-window.js (hunk 2 above)
Comment (miniak) [reply lib/browser/api/browser-window.js]: Btw, The second line of the comment block is not indented properly

**Comment by MarshallOfSound** [comment]
Closing this out as it gains conflicts too quickly for me to keep it up to date, this means that backports would be virtually impossible and I don't think that's something we want right now.  I'll try to find a way to make this transition smoother for a future attempt

**Pull request closed without merging**